caddyhttp/push/handler.go

```diff
@@ -62,9 +65,51 @@ outer:
 
 		target := strings.TrimSuffix(strings.TrimPrefix(parts[0], "<"), ">")
 
-		err := pusher.Push(target, &http.PushOptions{Method: http.MethodGet})
+		err := pusher.Push(target, &http.PushOptions{
+			Method: http.MethodGet,
+			Header: headers,
+		})
+
 		if err != nil {
 			break outer
 		}
 	}
 }
+
+func (h Middleware) mergeHeaders(l, r http.Header) http.Header {
+
+	out := http.Header{}
+
+	for k, v := range l {
+		out[k] = v
+	}
+
+	for k, vv := range r {
+		for _, v := range vv {
+			out.Add(k, v)
+		}
+	}
+
+	return out
+}
+
+func (h Middleware) filterProxiedHeaders(headers http.Header) http.Header {
+
+	filter := http.Header{}
+
+	for _, header := range proxiedHeaders {
+		if val, ok := headers[http.CanonicalHeaderKey(header)]; ok {
```

Comment:
It looks like the header names are already in the canonical format, as defined below.